service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java

```diff
@@ -20,28 +22,63 @@
+    public SampleManifestEntity saveSampleManifest(SampleManifestEntity 
+            sampleManifestEntity) throws Exception {
+        SampleManifestEntity updatedSampleManifestEntity = setUpSampleManifest(sampleManifestEntity);
         SampleManifestEntity foundSample =
-                sampleManifestRepository.findSampleByIgoId(sampleManifestEntity.getSampleIgoId());
+                sampleManifestRepository.findSampleByIgoId(updatedSampleManifestEntity.getSampleIgoId());
         if (foundSample == null) {
             PatientMetadata patient = patientMetadataRepository.findPatientByInvestigatorId(
-                    sampleManifestEntity.getPatient().getInvestigatorPatientId());
+                    updatedSampleManifestEntity.getPatient().getInvestigatorPatientId());
             if (patient != null) {
-                sampleManifestEntity.setPatientUuid(patient.getUuid());
+                updatedSampleManifestEntity.setPatientUuid(patient.getUuid());
             }
-            sampleManifestRepository.save(sampleManifestEntity);
+            sampleManifestRepository.save(updatedSampleManifestEntity);
         } else {
-            sampleManifestEntity.setUuid(foundSample.getUuid());
+            updatedSampleManifestEntity.setUuid(foundSample.getUuid());
             sampleManifestRepository.updateSampleManifestJson(
-                    sampleManifestEntity.getSampleManifestJsonEntity(), foundSample.getUuid());
+                    updatedSampleManifestEntity.getSampleManifestJsonEntity(), foundSample.getUuid());
+        }
+       
+        return updatedSampleManifestEntity;
+    }
+    
+    @Override
+    public SampleManifestEntity setUpSampleManifest(SampleManifestEntity
+            sampleManifestEntity) throws Exception {
+        PatientMetadata patient = new PatientMetadata();
+        patient.setInvestigatorPatientId(sampleManifestEntity.getCmoPatientId());
+        sampleManifestEntity.setPatient(patient);
+
+        Sample igoId = new Sample();
+        igoId.setIdSource("igoId");
+        igoId.setSampleId(sampleManifestEntity.getIgoId());
+        sampleManifestEntity.addSample(igoId);
+
+        Sample investigatorId = new Sample();
+        investigatorId.setIdSource("investigatorId");
+        investigatorId.setSampleId(sampleManifestEntity.getInvestigatorSampleId());
+        sampleManifestEntity.addSample(investigatorId);
+
+        SampleManifestJsonEntity sampleJson = new SampleManifestJsonEntity();
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+        sampleJson.setCreationDate(String.valueOf(timestamp.getTime()));
+        sampleJson.setSampleManifestJson(sampleManifestEntity.toString());
+        sampleManifestEntity.setSampleManifestJsonEntity(sampleJson);
+        
+        //TODO: check SampleManifestEntity mapping
+        if (sampleManifestEntity.getTumorOrNormal().equals("Tumor")) {
```

Comment:
I would introduce an Enum in the SampleManifestEntity class which captures Tumor and Normal elements.